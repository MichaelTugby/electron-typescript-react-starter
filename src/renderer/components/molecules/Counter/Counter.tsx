/** @jsx jsx */
import { css, jsx } from "@emotion/react";
import type { FunctionComponent, HTMLAttributes } from "react";

import Button from "~/components/atoms/Button";

interface ICounterType extends HTMLAttributes<HTMLDivElement> {
  onDecrease?: () => void;
  onIncrease?: () => void;
}
const Counter: FunctionComponent<ICounterType> = ({
  children,
  onDecrease,
  onIncrease,
  ...props
}) => (
  <div className="counter" {...props}>
    <Button primary className="counter__decrease-btn" onClick={onDecrease}>
      -
    </Button>
    <span
      css={css`
        padding: 0 5px;
      `}
      className="counter__text"
    >
      {children}
    </span>
    <Button primary className="counter__increase-btn" onClick={onIncrease}>
      +
    </Button>
  </div>
);

export default Counter;
