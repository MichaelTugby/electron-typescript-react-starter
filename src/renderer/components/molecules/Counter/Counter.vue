<template>
  <div class="counter">
    <Button class="counter__decrease-button" primary @click="onDecrement"
      >-</Button
    >
    <span class="counter__text">
      <slot />
    </span>
    <Button class="counter__increase-button" primary @click="onIncrement"
      >+</Button
    >
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import Button from "~/renderer/components/atoms/Button";
export default defineComponent({
  components: { Button },
  emits: ["decrement", "increment"],
  setup(_, { emit }) {
    const onDecrement = () => emit("decrement");
    const onIncrement = () => emit("increment");
    return {
      onDecrement,
      onIncrement,
    };
  },
});
</script>

<style lang="scss" scoped>
.counter__text {
  padding: 0 5px;
  @media (prefers-color-scheme: dark) {
    color: white;
  }
  @media (prefers-color-scheme: light) {
    color: black;
  }
}
</style>
