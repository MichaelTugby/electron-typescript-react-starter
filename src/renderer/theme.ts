import "@emotion/react";

const theme = {
  colors: {
    primary: "lightblue",
  },
};

type MyTheme = typeof theme;

declare module "@emotion/react" {
  export interface Theme extends MyTheme {} // eslint-disable-line @typescript-eslint/no-empty-interface
}

export default theme;
