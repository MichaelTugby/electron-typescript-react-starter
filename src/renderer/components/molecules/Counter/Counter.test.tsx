import React from "react";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "@emotion/react";
import Counter from "./Counter";

import theme from "~/theme";

describe("Counter", () => {
  afterEach(cleanup);

  it("snapshot has not changed", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Counter>1</Counter>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
