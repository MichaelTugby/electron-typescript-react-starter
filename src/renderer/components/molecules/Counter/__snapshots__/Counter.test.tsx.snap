// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot has not changed 1`] = `
<div>
  <div
    class="counter"
  >
    <button
      class="counter__decrease-btn css-1msykvu-Button"
    >
      -
    </button>
    <span
      class="counter__text css-uwz46b-Counter"
    >
      1
    </span>
    <button
      class="counter__increase-btn css-1msykvu-Button"
    >
      +
    </button>
  </div>
</div>
`;
